src/main/java/de/tum/in/www1/artemis/service/LearningPathService.java

```diff
@@ -178,17 +181,46 @@ private void updateLearningPathProgress(@NotNull LearningPath learningPath) {
      */
     public LearningPathHealthDTO getHealthStatusForCourse(@NotNull Course course) {
         if (!course.getLearningPathsEnabled()) {
-            return new LearningPathHealthDTO(LearningPathHealthDTO.HealthStatus.DISABLED);
+            return new LearningPathHealthDTO(Set.of(LearningPathHealthDTO.HealthStatus.DISABLED));
+        }
+
+        Set<LearningPathHealthDTO.HealthStatus> status = new HashSet<>();
+        Long numberOfMissingLearningPaths = checkMissingLearningPaths(course, status);
+        checkNoCompetencies(course, status);
+        checkNoRelations(course, status);
+
+        // if no issues where found, add OK status
+        if (status.isEmpty()) {
+            status.add(LearningPathHealthDTO.HealthStatus.OK);
         }
 
+        return new LearningPathHealthDTO(status, numberOfMissingLearningPaths);
+    }
+
+    private Long checkMissingLearningPaths(@NotNull Course course, @NotNull Set<LearningPathHealthDTO.HealthStatus> status) {
         long numberOfStudents = userRepository.countUserInGroup(course.getStudentGroupName());
         long numberOfLearningPaths = learningPathRepository.countLearningPathsOfEnrolledStudentsInCourse(course.getId());
+        Long numberOfMissingLearningPaths = numberOfStudents - numberOfLearningPaths;
```

Comment:
Can this be a long?

Reply:
`numberOfMissingLearningPaths ` is of type `Long` so it can be set to `null` if it is `0`. The method is called to construct a DTO containing the returned value. I added an annotation so the transferred json of the DTO will only contain fields that are not null to avoid sending irrelevant data.